suite-native/device-manager/src/components/DeviceItemContent.tsx

```diff
@@ -1,82 +1,201 @@
+                        <ConnectionDot isConnected={!!device?.connected} />
+
+                        <Text
+                            variant="hint"
+                            color={device.connected ? 'textSecondaryHighlight' : 'textSubdued'}
+                        >
+                            <Translation
+                                id={
+                                    device.connected
+                                        ? 'deviceManager.status.connected'
+                                        : 'deviceManager.status.disconnected'
+                                }
+                            />
+                        </Text>
+                    </HStack>
+                )}
+            </Box>
+        )}
+    </>
+);
+
+const WalletDetailDeviceItemContent = ({
+    device,
+    headerTextVariant,
+    isPortfolioLabelDisplayed,
+    header,
+    subHeader,
+    isPortfolioTrackerDevice,
+}: SwitchContentType) => (
+    <>
+        <HStack alignItems="center">
+            <Text variant={headerTextVariant}>{header}</Text>
+            {!isPortfolioTrackerDevice && device && (
+                <ConnectionDot isConnected={!!device?.connected} />
+            )}
+        </HStack>
+        {device?.id && (
+            <Box>
+                {isPortfolioTrackerDevice && isPortfolioLabelDisplayed && (
+                    <Text variant="label" color="textSubdued">
+                        <Translation id="deviceManager.status.portfolioTracker" />
+                    </Text>
+                )}
+                {!isPortfolioTrackerDevice && (
+                    <HStack alignItems="center" spacing="small">
+                        <Text variant="label" color="textSubdued">
+                            {subHeader}
+                        </Text>
+                    </HStack>
+                )}
+            </Box>
+        )}
+    </>
+);
+
 export const DeviceItemContent = ({
-    deviceId,
+    device,
     isPortfolioLabelDisplayed = true,
     headerTextVariant = 'body',
+    iconSize = 'large',
+    variant = 'simple',
 }: DeviceItemContentProps) => {
+    const { translate } = useTranslate();
+
     const deviceName = useSelector((state: DeviceRootState) =>
-        selectDeviceNameById(state, deviceId),
+        selectDeviceNameById(state, device?.id),
     );
+
     const deviceLabel = useSelector((state: DeviceRootState) =>
-        selectDeviceLabelById(state, deviceId),
+        selectDeviceLabelById(state, device?.id),
     );
 
-    const isPortfolioTrackerDevice = deviceId === PORTFOLIO_TRACKER_DEVICE_ID;
+    const isPortfolioTrackerDevice = device?.id === PORTFOLIO_TRACKER_DEVICE_ID;
+
+    const deviceHeader =
+        (isPortfolioTrackerDevice ? deviceName : deviceLabel) ??
+        translate('deviceManager.defaultHeader');
+
+    const trimmedHeader =
+        deviceHeader.length > 24 ? `${deviceHeader.slice(0, 24)}...` : deviceHeader;
```

Comment:
Why don't use use this? 
https://reactnative.dev/docs/text#ellipsizemode

Reply:
5825a46c34